openhands/integrations/bitbucket/bitbucket_service.py

```diff
@@ -385,6 +393,129 @@ async def create_pr(
+            )
+            raise ResourceNotFoundError(
+                f'Main branch not found for repository {repository}. '
+                f'This repository may be empty or have no default branch configured.'
+            )
+
+        # Step 2: Check for .cursorrules file
+        try:
+            cursorrules_url = f'{self.BASE_URL}/repositories/{repository}/src/{repo_details.main_branch}/.cursorrules'
+            cursorrules_response, _ = await self._make_request(cursorrules_url)
+            if cursorrules_response:
+                microagents.append(
+                    self._create_microagent_response('.cursorrules', '.cursorrules')
+                )
+        except ResourceNotFoundError:
+            logger.debug(f'No .cursorrules file found in {repository}')
+        except Exception as e:
+            logger.warning(f'Error checking .cursorrules file in {repository}: {e}')
+
+        # Step 3: Check for microagents directory and process .md files
+        try:
+            src_url = f'{self.BASE_URL}/repositories/{repository}/src/{repo_details.main_branch}/{microagents_path}'
+            directory_data, _ = await self._make_request(src_url)
+
+            if 'values' in directory_data:
+                for item in directory_data['values']:
+                    if (
+                        item['type'] == 'commit_file'
+                        and item['path'].endswith('.md')
+                        and not item['path'].endswith('README.md')
+                    ):
+                        try:
+                            # Extract file name from path
+                            file_name = item['path'].split('/')[-1]
+                            microagents.append(
+                                self._create_microagent_response(
+                                    file_name, item['path']
+                                )
+                            )
+                        except Exception as e:
+                            logger.warning(
+                                f'Error processing microagent {item["path"]}: {str(e)}'
+                            )
+        except ResourceNotFoundError:
+            logger.info(
+                f'No microagents directory found in {repository} at {microagents_path}'
+            )
+        except Exception as e:
+            logger.warning(f'Error fetching microagents directory: {str(e)}')
+
+        return microagents
+
+    async def get_microagent_content(
+        self, repository: str, file_path: str
+    ) -> MicroagentContentResponse:
+        """Fetch individual file content from Bitbucket repository.
+
+        Args:
+            repository: Repository name in format 'workspace/repo_slug'
+            file_path: Path to the file within the repository
+
+        Returns:
+            MicroagentContentResponse with parsed content and triggers
+
+        Raises:
+            RuntimeError: If file cannot be fetched or doesn't exist
+        """
+        try:
```

Comment:
same here we don't need the try catch the provider handler will manage it

same stuff applies for the other service classes